src/Testcontainers.MongoDb/MongoDbBuilder.cs

```diff
@@ -66,12 +70,19 @@ public override MongoDbContainer Build()
     /// <inheritdoc />
     protected override MongoDbBuilder Init()
     {
-        return base.Init()
+        var builder = base.Init()
```

Comment:
What are your thoughts on something like that:

```csharp
string username;
string password;

if (_useDefaultCredentials)
{
    username = DefaultUsername;
    password = DefaultPassword;
}
else
{
    username = null;
    password = null;
}

...
```

This approach keeps the builder configuration in one place. I don't have a strong opinion on this; I just considered it might be beneficial to maintain the configuration in a cohesive manner.

Reply:
This seems okay to me!

The only thing is I'm not sure how mongodb will behave if it has the environment variables available with no value; there might need to be a not-null check inside the WithUsername and WithPassword methods

I will update the PR later today

Reply:
I confirmed that `.WithEnvironment("MONGO_INITDB_ROOT_USERNAME", null)` and `.WithEnvironment("MONGO_INITDB_ROOT_PASSWORD", null)` cause no problems :)